src/snmalloc/global/memcpy.h

```diff
@@ -12,6 +12,17 @@ namespace snmalloc
   SNMALLOC_FAST_PATH_INLINE void copy_one(void* dst, const void* src)
   {
 #if __has_builtin(__builtin_memcpy_inline)
+    if constexpr (aal_supports<StrictProvenance>)
+    {
+      static_assert(Size <= sizeof(void*));
+      if constexpr (Size == sizeof(void*))
+      {
+        auto dp = static_cast<void**>(dst);
+        auto sp = static_cast<void* const*>(src);
+        *dp = *sp;
+        return;
```

Comment:
I don't think that this is correct, for two reasons:

 - It will not to capability-sized copies for larger-than-a-capability things.
 - It assumes capability alignment, which may not be the case.